<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Connecting data to Open Tree trees • rotl</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<link href="../ropensci.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../tidyverse-2.css" rel="stylesheet">
<meta property="og:title" content="Connecting data to Open Tree trees">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">rotl</a>
        <div class="info">
          <span class="partof">part of  <a href="https://ropensci.org"><img src="https://d33wubrfki0l68.cloudfront.net/88338b7637e084df5c43e7a9e6293719dbaf10e2/30903/img/icon_lettering_white.svg" height="24" alt="rOpenSci"></a></span>
          <span class="version label label-default version-default" data-toggle="tooltip" data-placement="bottom" title="Released version">3.0.7</span>
        </div>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/rotl.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/data_mashups.html">Connecting data to Open Tree trees</a>
    </li>
    <li>
      <a href="../articles/meta-analysis.html">Using the Open Tree synthesis in a comparative analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
        <li>
  <a href="https://github.com/ropensci/rotl">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Connecting data to Open Tree trees</h1>
                        <h4 class="author">David Winter</h4>
            
            <h4 class="date">2019-03-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/rotl/blob/master/vignettes/data_mashups.Rmd"><code>vignettes/data_mashups.Rmd</code></a></small>
      <div class="hidden name"><code>data_mashups.Rmd</code></div>

    </div>

    
    
<div id="combining-data-from-otol-and-other-sources-" class="section level2">
<h2 class="hasAnchor">
<a href="#combining-data-from-otol-and-other-sources-" class="anchor"></a>Combining data from OToL and other sources.</h2>
<p>One of the major goals of <code>rotl</code> is to help users combine data from other sources with the phylogenetic trees in the Open Tree database. This examples document describes some of the ways in which a user might connect data to trees from Open Tree.</p>
</div>
<div id="get-open-tree-ids-to-match-your-data-" class="section level2">
<h2 class="hasAnchor">
<a href="#get-open-tree-ids-to-match-your-data-" class="anchor"></a>Get Open Tree IDs to match your data.</h2>
<p>Let’s say you have a dataset where each row represents a measurement taken from one species, and your goal is to put these measurements in some phylogenetic context. Here’s a small example: the best estimate of the mutation rate for a set of unicellular Eukaryotes along with some other property of those species which might explain the mutation rate:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">csv_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"protist_mutation_rates.csv"</span>, <span class="dt">package =</span> <span class="st">"rotl"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">mu &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.csv</a></span>(csv_path, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">mu</a></code></pre></div>
<pre><code>##                     species       mu pop.size genome.size
## 1   Tetrahymena thermophila 7.61e-12 1.12e+08    1.04e+08
## 2    Paramecium tetraurelia 1.94e-11 1.24e+08    7.20e+07
## 3 Chlamydomonas reinhardtii 2.08e-10 1.00e+08    1.12e+08
## 4  Dictyostelium discoideum 2.90e-11 7.40e+06    3.40e+07
## 5  Saccharomyces cerevisiae 3.30e-10 1.00e+08    1.25e+08
## 6       Saccharomyces pombe 2.00e-10 1.00e+07    1.25e+08</code></pre>
<p>If we want to get a tree for these species we need to start by finding the unique ID for each of these species in the Open Tree database. We can use the Taxonomic Name Resolution Service (<code>tnrs</code>) functions to do this. Before we do that we should see if any of the taxonomic contexts, which can be used to narrow a search and avoid conflicts between different codes, apply to our group of species:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(rotl)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="../reference/tnrs_contexts.html">tnrs_contexts</a></span>()</a></code></pre></div>
<pre><code>## Possible contexts:
##    Animals 
##       Birds, Tetrapods, Mammals, Amphibians, Vertebrates 
##       Arthropods, Molluscs, Nematodes, Platyhelminthes, Annelids 
##       Cnidarians, Arachnids, Insects 
##    Bacteria 
##       SAR group, Archaea, Excavata, Amoebozoa, Centrohelida 
##       Haptophyta, Apusozoa, Diatoms, Ciliates, Forams 
##    Fungi 
##       Basidiomycetes, Ascomycetes 
##    Land plants 
##       Hornworts, Mosses, Liverworts, Vascular plants, Club mosses 
##       Ferns, Seed plants, Flowering plants, Monocots, Eudicots 
##       Rosids, Asterids, Asterales, Asteraceae, Aster 
##       Symphyotrichum, Campanulaceae, Lobelia 
##    All life</code></pre>
<p>Hmm, none of those groups contain all of our species. In this case we can search using the <code>All life</code> context and the function <code>tnrs_match_names</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">taxon_search &lt;-<span class="st"> </span><span class="kw"><a href="../reference/tnrs_match_names.html">tnrs_match_names</a></span>(<span class="dt">names=</span>mu<span class="op">$</span>species, <span class="dt">context_name=</span><span class="st">"All life"</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(taxon_search)</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">search_string</th>
<th align="left">unique_name</th>
<th align="left">approximate_match</th>
<th align="right">ott_id</th>
<th align="left">is_synonym</th>
<th align="left">flags</th>
<th align="right">number_matches</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">tetrahymena thermophila</td>
<td align="left">Tetrahymena thermophila</td>
<td align="left">FALSE</td>
<td align="right">180195</td>
<td align="left">FALSE</td>
<td align="left">SIBLING_HIGHER</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">paramecium tetraurelia</td>
<td align="left">Paramecium tetraurelia</td>
<td align="left">FALSE</td>
<td align="right">568130</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">chlamydomonas reinhardtii</td>
<td align="left">Chlamydomonas reinhardtii</td>
<td align="left">FALSE</td>
<td align="right">33153</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">dictyostelium discoideum</td>
<td align="left">Dictyostelium discoideum</td>
<td align="left">FALSE</td>
<td align="right">160850</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">saccharomyces cerevisiae</td>
<td align="left">Saccharomyces cerevisiae</td>
<td align="left">FALSE</td>
<td align="right">356221</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">saccharomyces pombe</td>
<td align="left">Schizosaccharomyces pombe</td>
<td align="left">FALSE</td>
<td align="right">990004</td>
<td align="left">TRUE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>Good, all of our species are known to Open Tree. Note, though, that one of the names is a synonym. <em>Saccharomyces pombe</em> is older name for what is now called <em>Schizosaccharomyces pombe</em>. As the name suggests, the Taxonomic Name Resolution Service is designed to deal with these problems (and similar ones like misspellings), but it is always a good idea to check the results of <code>tnrs_match_names</code> closely to ensure the results are what you expect.</p>
<p>In this case we have a good ID for each of our species so we can move on. Before we do that, let’s ensure we can match up our original data to the Open Tree names and IDs by adding them to our <code>data.frame</code>:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">mu<span class="op">$</span>ott_name &lt;-<span class="st"> </span><span class="kw"><a href="../reference/taxonomy-methods.html">unique_name</a></span>(taxon_search)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">mu<span class="op">$</span>ott_id &lt;-<span class="st"> </span>taxon_search<span class="op">$</span>ott_id</a></code></pre></div>
</div>
<div id="find-a-tree-with-your-taxa" class="section level2">
<h2 class="hasAnchor">
<a href="#find-a-tree-with-your-taxa" class="anchor"></a>Find a tree with your taxa</h2>
<p>Now let’s find a tree. There are two possible options here: we can search for published studies that include our taxa or we can use the ‘synthetic tree’ from Open Tree. We can try both approaches.</p>
<div id="published-trees" class="section level3">
<h3 class="hasAnchor">
<a href="#published-trees" class="anchor"></a>Published trees</h3>
<p>Before we can search for published studies or trees, we should check out the list of properties we can use to perform such searches:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="../reference/studies_properties.html">studies_properties</a></span>()</a></code></pre></div>
<pre><code>## $tree_properties
##  [1] "ot:treebaseOTUId"           "ot:nodeLabelMode"          
##  [3] "ot:originalLabel"           "oti_tree_id"               
##  [5] "ot:ottTaxonName"            "ot:inferenceMethod"        
##  [7] "ot:tag"                     "ot:comment"                
##  [9] "ot:treebaseTreeId"          "ot:branchLengthDescription"
## [11] "ot:treeModified"            "ot:studyId"                
## [13] "ot:branchLengthTimeUnits"   "ot:ottId"                  
## [15] "is_deprecated"              "ot:branchLengthMode"       
## [17] "ot:treeLastEdited"          "ot:nodeLabelDescription"   
## 
## $study_properties
##  [1] "ot:studyModified"             "ot:focalClade"               
##  [3] "ot:focalCladeOTTTaxonName"    "ot:focalCladeOTTId"          
##  [5] "ot:studyPublication"          "ot:studyLastEditor"          
##  [7] "ot:tag"                       "ot:focalCladeTaxonName"      
##  [9] "ot:comment"                   "ot:studyLabel"               
## [11] "ot:authorContributed"         "ot:studyPublicationReference"
## [13] "ot:curatorName"               "ot:studyId"                  
## [15] "ot:studyUploaded"             "ot:studyYear"                
## [17] "is_deprecated"                "ot:dataDeposit"              
## [19] "ot:candidateTreeForSynthesis"</code></pre>
<p>We have <code>ottIds</code> for our taxa, so let’s use those IDs to search for trees that contain them. Starting with our first species <em>Tetrahymena thermophila</em> we can use <code>studies_find_trees</code> to do this search.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="../reference/studies_find_trees.html">studies_find_trees</a></span>(<span class="dt">property=</span><span class="st">"ot:ottId"</span>, <span class="dt">value=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(<span class="kw"><a href="../reference/taxonomy-methods.html">ott_id</a></span>(taxon_search)[<span class="dv">1</span>]))</a></code></pre></div>
<pre><code>## [1] study_ids       dat             n_matched_trees
## &lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>Well… that’s not very promising. We can repeat that process for all of the IDs to see if the other species are better represented.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">hits &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(mu<span class="op">$</span>ott_id, studies_find_trees, <span class="dt">property=</span><span class="st">"ot:ottId"</span>, <span class="dt">detailed =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">sapply</a></span>(hits, <span class="cf">function</span>(x) <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(x[[<span class="st">"n_matched_trees"</span>]]))</a></code></pre></div>
<pre><code>## [1] 0 0 4 0 5 3</code></pre>
<p>OK, most of our species are not in any of the published trees available. You can help fix this sort of problem by <a href="https://tree.opentreeoflife.org/curator">making sure you submit your published trees to Open Tree</a>.</p>
</div>
<div id="a-part-of-the-synthesis-tree" class="section level3">
<h3 class="hasAnchor">
<a href="#a-part-of-the-synthesis-tree" class="anchor"></a>A part of the synthesis tree</h3>
<p>Thankfully, we can still use the complete Tree of Life made from the combined results of all of the published trees and taxonomies that go into Open Tree. The function <code>tol_induced_subtree</code> will fetch a tree relating a set of IDs.</p>
<p>Using the default arguments you can get a tree object into your R session:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">ott_in_tree &lt;-<span class="st"> </span><span class="kw"><a href="../reference/taxonomy-methods.html">ott_id</a></span>(taxon_search)[<span class="kw"><a href="../reference/is_in_tree.html">is_in_tree</a></span>(<span class="kw"><a href="../reference/taxonomy-methods.html">ott_id</a></span>(taxon_search))]</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">tr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/tol_induced_subtree.html">tol_induced_subtree</a></span>(<span class="dt">ott_ids =</span> ott_in_tree)</a></code></pre></div>
<pre><code>## Warning in collapse_singles(tr, show_progress): Dropping singleton
## nodes with labels: mrcaott2ott276, mrcaott2ott142555, mrcaott2ott3043,
## mrcaott2ott7623, Chloroplastida ott361838, Chlorophyta ott979501,
## mrcaott185ott23323, mrcaott185ott599, mrcaott1427ott25671,
## mrcaott1427ott1544, mrcaott1544ott15345, mrcaott1544ott1795,
## mrcaott9389ott365252, mrcaott9389ott23557, mrcaott23557ott527099,
## mrcaott149ott902, SAR ott5246039, Alveolata ott266751, Ciliophora (phylum
## in subkingdom SAR) ott302424, Intramacronucleata ott340382, Conthreep
## ott5248773, Peniculia ott5257364, Paramecium (genus in subkingdom SAR)
## ott568126, Hymenostomatia ott5257367, Tetrahymena ott47284, Amoebozoa
## ott1064655, mrcaott3973ott5765, mrcaott3973ott26103, mrcaott26103ott273110,
## mrcaott26103ott229626, Dictyostelia ott835575, Dictyostelium ott999665,
## Opisthokonta ott332573, Nucletmycea ott5246132, Fungi ott352914,
## mrcaott48ott3465, mrcaott48ott67172, mrcaott48ott1423, mrcaott48ott4814,
## h2007-2 ott5576447, h2007-1 ott5584405, Dikarya ott656316,
## mrcaott48ott1053, Saccharomycetales ott5976840, Saccharomycetaceae
## ott989994, Saccharomyces ott908546, Taphrinomycotina ott656312,
## Schizosaccharomycetes ott921286, Schizosaccharomycetidae ott5670481,
## Schizosaccharomycetales ott508517, Schizosaccharomycetaceae ott990009,
## Schizosaccharomyces ott990008</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tr)</a></code></pre></div>
<p><img src="data_mashups_files/figure-html/subtree-1.png" width="672"></p>
</div>
<div id="connect-your-data-to-the-tips-of-your-tree" class="section level3">
<h3 class="hasAnchor">
<a href="#connect-your-data-to-the-tips-of-your-tree" class="anchor"></a>Connect your data to the tips of your tree</h3>
<p>Now we have a tree for of our species, how can we use the tree and the data together?</p>
<p>The package <code>phylobase</code> provide an object class called <code>phylo4d</code>, which is designed to represent a phylogeny and data associated with its tips. In oder to get our tree and data into one of these objects we have to make sure the labels in the tree and in our data match exactly. That’s not quite the case at the moment (tree labels have underscores and IDs appended):</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">mu<span class="op">$</span>ott_name[<span class="dv">1</span>]</a></code></pre></div>
<pre><code>## $`Tetrahymena thermophila`
## [1] "Tetrahymena thermophila"</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">tr<span class="op">$</span>tip.label[<span class="dv">4</span>]</a></code></pre></div>
<pre><code>## [1] "Dictyostelium_discoideum_ott160850"</code></pre>
<p><code>rotl</code> provides a convienence function <code>strip_ott_ids</code> to deal with these.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">tr<span class="op">$</span>tip.label &lt;-<span class="st"> </span><span class="kw"><a href="../reference/strip_ott_ids.html">strip_ott_ids</a></span>(tr<span class="op">$</span>tip.label, <span class="dt">remove_underscores=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">tr<span class="op">$</span>tip.label <span class="op">%in%</span><span class="st"> </span>mu<span class="op">$</span>ott_name</a></code></pre></div>
<pre><code>## [1] TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<p>Ok, now the tips are together we can make a new dataset. The <code><a href="https://www.rdocumentation.org/packages/phylobase/topics/phylo4d-methods">phylo4d()</a></code> functions matches tip labels to the row names of a <code>data.frame</code>, so let’s make a new dataset that contains just the relevant data and has row names to match the tree</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(phylobase)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2">mu_numeric &lt;-<span class="st"> </span>mu[, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"mu"</span>, <span class="st">"pop.size"</span>, <span class="st">"genome.size"</span>)]</a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(mu_numeric) &lt;-<span class="st"> </span>mu<span class="op">$</span>ott_name</a>
<a class="sourceLine" id="cb22-4" data-line-number="4">tree_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/phylobase/topics/phylo4d-methods">phylo4d</a></span>(tr, mu_numeric)</a></code></pre></div>
<p>And now we can plot the data and the tree together</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/phylobase/topics/treePlot-methods">plot</a></span>(tree_data)</a></code></pre></div>
<p><img src="data_mashups_files/figure-html/unnamed-chunk-1-1.png" width="672"></p>
</div>
</div>
<div id="find-external-data-associated-with-studies-trees-and-taxa-from-open-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#find-external-data-associated-with-studies-trees-and-taxa-from-open-tree" class="anchor"></a>Find external data associated with studies, trees and taxa from Open Tree</h2>
<p>In the above example we looked for a tree that related species in another dataset. Now we will go the other way, and try to find data associated with Open Tree records in other databases.</p>
<div id="get-external-data-from-a-study" class="section level3">
<h3 class="hasAnchor">
<a href="#get-external-data-from-a-study" class="anchor"></a>Get external data from a study</h3>
<p>Let’s imagine you were interested in extending or reproducing the results of a published study. If that study is included in Open Tree you can find it via <code>studies_find_studies</code> or <code>studies_find_trees</code> and retrieve the published trees with <code>get_study</code>. <code>rotl</code> will also help you find external. The function <code>study_external_IDs</code> retrieves the DOI for a given study, and uses that to gather some more data:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">extra_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/try">try</a></span>(<span class="kw"><a href="../reference/study_external_IDs.html">study_external_IDs</a></span>(<span class="st">"pg_1980"</span>), <span class="dt">silent=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/class">inherits</a></span>(extra_data, <span class="st">"try-error"</span>))</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">  extra_data</a></code></pre></div>
<pre><code>## External data identifiers for study 
##  $doi:  10.1016/j.ympev.2006.04.016 
##  $pubmed_id:  16762568 
##  $nucleotide_ids: vector of 58 IDs
##  $external_data_url http://purl.org/phylo/treebase/phylows/study/TB2:S1575</code></pre>
<p>Here the returned object contains an <code>external_data_url</code> (in this case a link to the study in Treebase), a pubmed ID for the paper and a vector IDs for the NCBI’s nuleotide database. The packages <code>treebase</code> and <code>rentrez</code> provide functions to make use of these IDs within R.</p>
<p>As an example, let’s use <code>rentrez</code> to download the first two DNA seqences and print them.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(rentrez)</a>
<a class="sourceLine" id="cb26-2" data-line-number="2">seqs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/try">try</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/rentrez/topics/entrez_fetch">entrez_fetch</a></span>(<span class="dt">db=</span><span class="st">"nucleotide"</span>, <span class="dt">id=</span>extra_data<span class="op">$</span>nucleotide_ids[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="dt">rettype=</span><span class="st">"fasta"</span>), <span class="dt">silent =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb26-3" data-line-number="3"></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/class">inherits</a></span>(seqs, <span class="st">"try-error"</span>)) {</a>
<a class="sourceLine" id="cb26-5" data-line-number="5">  <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(<span class="st">"NCBI temporarily down."</span>)</a>
<a class="sourceLine" id="cb26-6" data-line-number="6">} <span class="cf">else</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(seqs)</a></code></pre></div>
<pre><code>## &gt;AM181011.1 Plectroninia neocaledoniense partial 28S rRNA gene, specimen voucher G316300 (Queensland Museum)
## GCTAGCAAGCGCGTCGGTGGTTCAGCCGGCTGGTCTCGTCGAGTTGTCGGTGTGCGGATCCGAACGGACC
## GCGGCCGATGGCGTCGGCGGGCAAGCTGTGGTGCACTCTGTCGGCGTGCGCGTCAGCGTCGGTTTCGGCC
## GGACGACGAGGCGCTCGGGGAAGGTAGCTGGACCGGTCTTCGGTGCAGTGTTATAGCCCTGGGCCGCTGG
## GTTCGGCGTTTGGGACCGAGGAGAGAGATGATCGCTGCAGCGCCTGTCTCCCTCTCGAGGGGGGCTAGCC
## AGCCGCTGTTTGGGTGGCGTCACTGGCGGAGGACTGCACGCAGTGCTTCGCCGGTGGTCGTGTCCAGGCG
## GGCGGTGTGGGTATAGAGGCGCTTAGGACGCTGGCGTCCAAATGGCCGTGCGCGACCCGTCTTGAAACAC
## GGACCAAGGAGTCTAGCATGTGCGCGAGTCTTAGGGTGTGGAAGCCCTCGGGCGCAATGAAAGTGAAGGG
## CCGTCGTCTCTCGGGGCTGCGGTGTGAGGTGAGAGCCGTCGCCGTCGGGTGGCGGTGCATCATCGGCCGG
## TCCATCCTGCTCTCAGGAGGATCTGCGCAAGAGCGTGTTTGCTGGGACCCGAAAGATGGTGAACTATGCC
## TGAATAGGGTGAAGCCAGAGGAAACTCTGGTGGAGGCTCGTAGCGGTTCTGACGTGCAAATCGATCGTCA
## AATTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTCCGAAGTTTCCCTCAGG
## ATAGCTGGAACTCGTCTTGACACAGTTTTATCAGGTAAAGCGAATGATTAGAGGTCTTGGGGGTGAAACA
## CCCTCAACCTATTCTCAAACTTTAAATAGGTAAGAAGCGCGACTTGCTCAATTGAAGTGGCGCGCAGTGA
## ATGTGAGTTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAACGCTCGGTTA
## AGGTGCCCAAGTCGACGCTCATCAGACCCCAGAAAAGGTGTTGGTCGATATAGACAGCAGGACGGTGGCC
## ATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCTGAAAATGGAT
## GGCGCTCAAGCGTCGCACCTATACCGAGCCGTCGTGGTAAATGCCAGGCCACGACGAGTAGGAGGGCGCG
## GTGGTCGTGACGCAGCCCTTGGCGCGAGCCTGGGCGAAACGGCCTCCGGTGCAGATCTTGGTGGTAGTAG
## CAAATATTCAAATGAGAGCTTTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGG
## GTTAGTCGATCCTAAGAGATAGGGAAGTTCCGTGTGAAAGTGCGCAATGCGCTTCTGTGCTGCGCGCCTC
## CTATCGAAAGGGAATCGGGTTAATATTCCCGAACCGGAAGGCGGATATCTCTGGCTCTCGGGTCAGGAGC
## GGCAACGCAAGCGTACTGCGAGACGTCGGCGGGGGCTCCGGGAAGAGTTGTCTTTTCTTTTTAACGCAGT
## CGCCATCCCTGGAATCGGTTTGCCCGGAGATAGGGTTGGCTGGCTCGGTAAAGCAGCACACTTCATGTGC
## TGTCCGGTGCGCTCTCGACGGCCCTTGAAAATCGCAGGTGTGCATCGATTCTCGCATCCGGTCGTACTCA
## TAACCGCATCAGGTCTCCAAGGT
## 
## &gt;AM181010.1 Eilhardia schulzei partial 28S rRNA gene, specimen voucher G316071 (Queensland Museum)
## GCTAGTAATGTACGTTGGTGGTTCAGCCGGCTAGTCTTGTCGAGTCGTCGTGTGGTGGATCCGACTGGAC
## CGTCCGCGGTGGTGTCGGCGGGCGAGCTGTGGTGCACTCTACGGACGTGCGCGTCAGCGTCGGTTCTCGA
## TGGGCGATAAGGTGCGTGGGGGAAGGTGGCTCGGTCCTTGGGAACTGAGTGTTACAGACCCTGGTGCTGG
## GCTCGTCGTGGGACCGAGGAGAGAGAGAGATGATCGCTGCGGCACCTGCCCCGTTGTCATTTTTCGGGGC
## TAGCCAGCCGTTTGTCAGGTGTGCGTCGGACGTTGAGGACTGCACGCAGTGCTGGACGTGGAGGCGTGAT
## CTGATGGCGGTGTGGGCATTAGAGGTGCCTAGGACGCTGGCGTCCAAATGGCCGTGCGCGACCCGTCTTG
## AAACACGGACCAAGGAGTCTAACATGTGCGCGAGTCTTAGGGTGTGCAAGCCCTCGGGCGCAATGAAAGT
## GAAGGCTCGGCGGCGCTAGTCGAGCTGAGGTGAGAGCCGTGGCCGTTGCATGTGGCGGCGGCGGCGCATC
## ATCGGCCGGTCCATCCTGCTCTCAGGGGGATCCGAGCAAGAGCGTATTTGTTGGGACCCGAAAGATGGTG
## AACTATGCCTGAATAGGGTGAAGCCAGAGGAAACTCTGGTGGAGGCTCGTAGCGATTCTGACGTGCAAAT
## CGATCGTCAAATTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTCCGAAGTT
## TCCCTCAGGATAGCTGGAGCTCTTGGACACAGTTTTATCAGGTAAAGCGAATGATCAGAGGTCTTGGGGG
## TGAAACACCCTCAACCTATTCTCAAACTTTAAATCGGTAAGAAGCGCGACTTGCTGAATTGAAGCCGCGC
## GCAAGCAATGTGAGTTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAACGC
## TGGGTTAAGGTGCCAAAGTCGACGCTCATCAGACCCCAGAAAAGGTGTTGGTTGATATAGACAGCAGGAC
## GATGGCCATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCTGAA
## AATGGATGGCGCTCAAGCGTCGCACCTATACCGGGCCGTCGTCGCAAATGCCAGGCGACGACGAGTAGGA
## GGGCGCAGTGGTCGTCATGCAGCCCTTGGCGTGAGCCTGGGTCAAACGGCCTCTGGTGCAGATCTTGGTG
## GTAGTAGCAAATATTCAAATGAGAGCTTTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTG
## GACATGGGTTAGTCGATCCTAAGTGATAGGGGAGCTCCGTATGAAAGTGCGCAATCGGCCCTGCTTGTGT
## CGCCTTGCGCCACCTATCGAAAGGGAATCGGGTTAATATTCCCGAACCGGAAGGCGGATTTTCTCTGGCT
## CTCGGGTCAGGAGCGGCAACGCTAGCGAACCGCGAGACGTCGGCGGGGGCTCCGGGAAGAGTTGTCTTTT
## CTTTTTAACGCAGTCGCCATCCCTGGAATCGGTTTGCCCGGAGATAGGGTTGGCTGGCTCGGTAAAGCAG
## CACACTTCATGTGCTGTCCGGTGCGCTCTCGACGGCCCTTGAAAATCGCGGCGAGTGTAGTCTGATTTTC
## GCATCCGTTCGTACTCATAACCGCATCAGGTCTCCAAGGT</code></pre>
<p>You could further process these sequences in R with the function <code>read.dna</code> from <code>ape</code> or save them to disk by specifying a file name with <code>cat</code>.</p>
</div>
<div id="find-a-ott-taxon-in-another-taxonomic-database" class="section level3">
<h3 class="hasAnchor">
<a href="#find-a-ott-taxon-in-another-taxonomic-database" class="anchor"></a>Find a OTT taxon in another taxonomic database</h3>
<p>It is also possible map an Open Tree taxon to a record in another taxonomic database. For instance, if we wanted to search for data about one of the tips of the sub-tree we fetched in the example above we could do so using <code>taxon_external_IDs</code>:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">Tt_ids &lt;-<span class="st"> </span><span class="kw"><a href="../reference/taxon_external_IDs.html">taxon_external_IDs</a></span>(mu<span class="op">$</span>ott_id[<span class="dv">2</span>])</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">Tt_ids</a></code></pre></div>
<pre><code>##   source       id
## 1  silva AY102613
## 2   ncbi     5888
## 3   gbif  7415807</code></pre>
<p>A user could then use <code>rgbif</code> to find locality records using the gbif ID or <code>rentrez</code> to get genetic or bibliometric data about from the NCBI’s databases.</p>
</div>
</div>
<div id="what-next" class="section level2">
<h2 class="hasAnchor">
<a href="#what-next" class="anchor"></a>What next</h2>
<p>The demonstration gets you to the point of visualizing your data in a phylogenetic context. But there’s a lot more you do with this sort of data in R. For instance, you could use packages like <code>ape</code>, <code>caper</code>, <code>phytools</code> and <code>mcmcGLMM</code> to perform phylogenetic comparative analyses of your data. You could gather more data on your species using packages that connect to trait databases like <code>rfishbase</code>, <code>AntWeb</code> or <code>rnpn</code> which provides data from the US National Phenology Network. You could also use <code>rentrez</code> to find genetic data for each of your species, and use that data to generate branch lengths for the phylogeny.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#combining-data-from-otol-and-other-sources-">Combining data from OToL and other sources.</a></li>
      <li><a href="#get-open-tree-ids-to-match-your-data-">Get Open Tree IDs to match your data.</a></li>
      <li><a href="#find-a-tree-with-your-taxa">Find a tree with your taxa</a></li>
      <li><a href="#find-external-data-associated-with-studies-trees-and-taxa-from-open-tree">Find external data associated with studies, trees and taxa from Open Tree</a></li>
      <li><a href="#what-next">What next</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div>
  <p>rotl is a part of <img src="https://github.com/ropensci/logos/raw/master/icon_lettering_color.png" height="40" alt="rOpenSci">. Learn more at <a href="https://ropensci.org">ropensci.org</a>.</p>
</div>

<div class="author">
  <p>
    Developed by Francois Michonneau, Joseph Brown, David Winter.
    Site built by <a href="https://pkgdown.r-lib.org">pkgdown</a>.
  </p>
</div>
      </footer>
</div>

  

  </body>
</html>
