<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using the Open Tree synthesis in a comparative analysis • rotl</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using the Open Tree synthesis in a comparative analysis">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rotl</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">3.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/data_mashups.html">Connecting data to Open Tree trees</a>
    </li>
    <li>
      <a href="../articles/how-to-use-rotl.html">How to use rotl?</a>
    </li>
    <li>
      <a href="../articles/meta-analysis.html">Using the Open Tree synthesis in a comparative analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/rotl">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Using the Open Tree synthesis in a comparative analysis</h1>
                        <h4 class="author">David Winter</h4>
            
            <h4 class="date">2019-01-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/rotl/blob/master/vignettes/meta-analysis.Rmd"><code>vignettes/meta-analysis.Rmd</code></a></small>
      <div class="hidden name"><code>meta-analysis.Rmd</code></div>

    </div>

    
    
<div id="phylogenetic-comparative-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogenetic-comparative-methods" class="anchor"></a>Phylogenetic Comparative Methods</h2>
<p>The development of phylogenetic comparative methods has made phylogenies and important source of data in fields as diverse as ecology, genomic and medicine. Comparative methods can be used to investigate patterns in the evolution of traits or the diversification of lineages. In other cases a phylogeny is treated as a “nuisance parameter”, allowing with the autocorrelation created by the shared evolutionary history of the different species included to be controlled for.</p>
<p>In many cases finding a tree that relates the species for which trait data are available is a rate-limiting step in such comparative analyses. Here we show how the synthetic tree provided by Open Tree of Life (and made available in R via <code>rotl</code>) can help to fill this gap.</p>
</div>
<div id="a-phylogenetic-meta-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#a-phylogenetic-meta-analysis" class="anchor"></a>A phylogenetic meta-analysis</h2>
<p>To demonstrate the use of <code>rotl</code> in a comparative analysis, we will partially reproduce the results of <a href="dx.doi.org/10.1111/jeb.12282">Rutkowska <em>et al</em> 2014</a>. Very briefly, this study is a meta-analysis summarising the results of multiple studies testing for systematic differences in the size of eggs which contain male and female offspring. Such a difference might mean that birds invest more heavily in one sex than the other.</p>
<p>Because this study involves data from 51 different species, Rutkowska <em>et al</em> used a phylogenetic comparative approach to account for the shared evolutionary history among some of the studied-species.</p>
<div id="gather-the-data" class="section level3">
<h3 class="hasAnchor">
<a href="#gather-the-data" class="anchor"></a>Gather the data</h3>
<p>If we are going to reproduce this analysis, we will first need to gather the data. Thankfully, the data is available as supplementary material from the publisher’s website. We can collect the data from using <code>fulltext</code> (with the papers DOI as input) and read it into memory with <code><a href="https://readxl.tidyverse.org/reference/read_excel.html">readxl::read_excel</a></code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(rotl)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">## This should work, but Wiley has currently broken the URLs to access the</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">## SI. </a>
<a class="sourceLine" id="cb2-3" data-line-number="3">## if (require(readxl) &amp;&amp; require(fulltext)) {</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">## doi &lt;- "10.1111/jeb.12282"</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">## xl_file &lt;- try(ft_get_si(doi, 1, save.name="egg.xls"), silent = TRUE)    </a>
<a class="sourceLine" id="cb2-6" data-line-number="6">## egg_data &lt;- read_excel(xl_file)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">## } else {</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">egg_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.csv</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"egg.csv"</span>, <span class="dt">package =</span> <span class="st">"rotl"</span>),</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">                     <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-10" data-line-number="10">## }</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(egg_data)</a></code></pre></div>
<pre><code>##                   animal                   Spp       Lndim Measure Neggs
## 1 Zonotrichia_leucophrys White-crowned sparrow 0.000000000  volume   294
## 2      Passer_domesticus         House sparrow 0.009407469  volume   149
## 3        Serinus_canaria                Canary 0.000000000  volume    52
## 4          Turdus_merula    European blackbird 0.021189299  volume    82
## 5    Agelaius_phoeniceus  Red-winged blackbird 0.218316086  volume   394
## 6    Quiscalus_mexicanus  Great-tailed grackle 0.281894985    mass   822
##   Nclutches        ESr Type         StudyID Year        D        EN
## 1        73 0.14004594 stat        Mead1987 1987 3.421918  85.91673
## 2        31 0.11175203 stat     Cordero2000 2000 4.045161  36.83413
## 3        21 0.49679140 stat     Leitner2006 2006 2.180952  23.84279
## 4        54 0.38598540 stat     Martyka2010 2010 1.414815  57.95812
## 5       106 0.07410136  raw Weatherhead1985 1985 3.173585 124.14982
## 6       205 0.05178834  raw     Teather1989 1989 3.407805 241.21099
##           Zr         VZr
## 1 0.14097244 0.012060292
## 2 0.11222075 0.029555954
## 3 0.54503712 0.047978211
## 4 0.40707397 0.018195675
## 5 0.07423744 0.008254242
## 6 0.05183471 0.004197959</code></pre>
<p>The most important variable in this dataset is <code>Zr</code>, which is a <a href="https://en.wikipedia.org/wiki/Fisher_transformation">normalized effect size</a> for difference ,in size between eggs that contain males and females. Values close to zero come from studies that found the sex of an egg’s inhabitant had little effect in its size, while large positive or negative values correspond to studies with substantial sex biases (towards males and females respectively). Since this is a meta-analysis we should produce the classic <a href="https://en.wikipedia.org/wiki/Funnel_plot">funnel plot</a> with effects-size on the y-axis and precision (the inverse of the sample standard error) on the x-axis. Here we calculate precision from the sample variance (<code>Vzr</code>):</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="dv">1</span><span class="op">/</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">sqrt</a></span>(egg_data<span class="op">$</span>VZr), egg_data<span class="op">$</span>Zr, <span class="dt">pch=</span><span class="dv">16</span>,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">     <span class="dt">ylab=</span><span class="st">"Effect size (Zr)"</span>,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">     <span class="dt">xlab=</span><span class="st">"Precision (1/SE)"</span>,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">     <span class="dt">main=</span><span class="st">"Effect sizes for sex bias in egg size among 51 brid species"</span> )</a></code></pre></div>
<p><img src="meta-analysis_files/figure-html/eggs_in_a_funnel-1.png" width="576"></p>
<p>In order to use this data later on we need to first convert it to a standard <code>data.frame</code>. We can also convert the <code>animal</code> column (the species names) to lower case which will make it easier to match names later on:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">egg_data &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(egg_data)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">egg_data<span class="op">$</span>animal &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/chartr">tolower</a></span>(egg_data<span class="op">$</span>animal)</a></code></pre></div>
</div>
<div id="find-the-species-in-ott" class="section level3">
<h3 class="hasAnchor">
<a href="#find-the-species-in-ott" class="anchor"></a>Find the species in OTT</h3>
<p>We can use the OTL synthesis tree to relate these species. To do so we first need to find Open Tree Taxonomy (OTT) IDs for each species. We can do that with the Taxonomic Name Resolution Service function <code>tnrs_match_names</code>:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">taxa &lt;-<span class="st"> </span><span class="kw"><a href="../reference/tnrs_match_names.html">tnrs_match_names</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(egg_data<span class="op">$</span>animal), <span class="dt">context=</span><span class="st">"Animals"</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(taxa)</a></code></pre></div>
<pre><code>##            search_string            unique_name approximate_match ott_id
## 1 zonotrichia_leucophrys Zonotrichia leucophrys              TRUE 265553
## 2      passer_domesticus      Passer domesticus              TRUE 745175
## 3        serinus_canaria        Serinus canaria              TRUE 464865
## 4          turdus_merula          Turdus merula              TRUE 568572
## 5    agelaius_phoeniceus    Agelaius phoeniceus              TRUE 226605
## 6    quiscalus_mexicanus    Quiscalus mexicanus              TRUE 743411
##   is_synonym          flags number_matches
## 1      FALSE                             1
## 2      FALSE                             1
## 3      FALSE SIBLING_HIGHER              2
## 4      FALSE                             1
## 5      FALSE                             2
## 6      FALSE                             1</code></pre>
<p>All of these species are in OTT, but a few of them go by different names in the Open Tree than we have in our data set. Because the tree <code>rotl</code> fetches will have Open Tree names, we need to create a named vector that maps the names we have for each species to the names Open Tree uses for them:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">taxon_map &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/structure">structure</a></span>(taxa<span class="op">$</span>search_string, <span class="dt">names=</span>taxa<span class="op">$</span>unique_name)</a></code></pre></div>
<p>Now we can use this map to retrieve “data set names” from “OTT names”:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">taxon_map[<span class="st">"Anser caerulescens"</span>]</a></code></pre></div>
<pre><code>##  Anser caerulescens 
## "chen_caerulescens"</code></pre>
</div>
<div id="get-a-tree" class="section level3">
<h3 class="hasAnchor">
<a href="#get-a-tree" class="anchor"></a>Get a tree</h3>
<p>Now we can get the tree. There are really too many tips here to show nicely, so we will leave them out of this plot</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">tr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/tol_induced_subtree.html">tol_induced_subtree</a></span>(<span class="kw"><a href="../reference/taxonomy-methods.html">ott_id</a></span>(taxa)[<span class="kw"><a href="../reference/is_in_tree.html">is_in_tree</a></span>(<span class="kw"><a href="../reference/taxonomy-methods.html">ott_id</a></span>(taxa))])</a></code></pre></div>
<pre><code>## Warning in collapse_singles(tr, show_progress): Dropping singleton nodes
## with labels: mrcaott24ott6612, Lophotrochozoa ott155737, mrcaott24ott56,
## mrcaott56ott5497, Mollusca ott802117, mrcaott56ott1881, mrcaott56ott1589,
## Bivalvia ott1025545, mrcaott2556ott10806, mrcaott2556ott6150, Pteriomorphia
## ott1025543, mrcaott2556ott2567, mrcaott2567ott26482, Pterioida ott307066,
## Pterioidea ott713805, Pteriidae ott645953, Pteria ott892317, Ecdysozoa
## ott611099, Panarthropoda ott816442, Arthropoda ott632179, Mandibulata
## ott985907, Pancrustacea ott985906, mrcaott87ott704, mrcaott704ott4932,
## Copepoda ott461528, Neocopepoda ott754178, mrcaott4932ott12421,
## mrcaott12421ott92630, mrcaott12421ott14254, Harpacticoida ott604252,
## mrcaott14254ott2953829, mrcaott14254ott296853, mrcaott14254ott535412,
## mrcaott14254ott14257, mrcaott14254ott14260, mrcaott14254ott129252,
## mrcaott14254ott41175, mrcaott14254ott586975, Harpacticidae ott639300, Zaus
## ott14256, Deuterostomia ott147604, Chordata ott125642, mrcaott42ott658,
## Craniata (subphylum in Deuterostomia) ott947318, Vertebrata (subphylum
## in Opisthokonta) ott801601, Gnathostomata (superclass in phylum
## Chordata) ott278114, Teleostomi ott114656, Euteleostomi ott114654,
## Sarcopterygii ott458402, Dipnotetrapodomorpha ott4940726, Tetrapoda
## ott229562, Amniota ott229560, Sauropsida ott639642, Sauria (order
## in Opisthokonta) ott329823, Archelosauria ott4947372, Archosauria
## ott335588, Dinosauria ott90215, Coelurosauria ott664351, Theropoda
## ott664350, Saurischia ott90223, Aves ott81461, mrcaott246ott5481,
## mrcaott246ott7145, mrcaott246ott928360, mrcaott246ott1858,
## mrcaott246ott2907, mrcaott246ott3600042, mrcaott246ott7113, Passeriformes
## ott1041547, mrcaott246ott3212, mrcaott246ott428578, mrcaott246ott44866,
## mrcaott246ott5929, mrcaott246ott32658, mrcaott246ott22325,
## mrcaott246ott176461, mrcaott246ott10351, mrcaott246ott5934,
## mrcaott246ott1566, mrcaott1566ott3598440, mrcaott1566ott496009,
## mrcaott1566ott35326, mrcaott1566ott32651, mrcaott1566ott20498,
## mrcaott1566ott22300, mrcaott22300ott35350, mrcaott22300ott547548,
## mrcaott22300ott909199, mrcaott22300ott67150, mrcaott22300ott130294,
## mrcaott22300ott3598245, mrcaott22300ott416087, mrcaott22300ott629342,
## mrcaott22300ott416089, mrcaott22300ott107840, mrcaott113980ott364210,
## mrcaott364210ott3598839, mrcaott364210ott5341363, mrcaott19467ott431648,
## mrcaott19467ott252687, mrcaott19467ott482248, mrcaott19467ott401023,
## mrcaott19467ott233446, mrcaott19467ott168078, mrcaott19467ott1046624,
## mrcaott19467ott161293, mrcaott19467ott46396, mrcaott46396ott46398,
## mrcaott46398ott168083, mrcaott168083ott431676, mrcaott2175ott968664,
## mrcaott2175ott259082, mrcaott2175ott59905, mrcaott2175ott2224,
## mrcaott2224ott366470, mrcaott3364ott73828, mrcaott3364ott4083,
## mrcaott4083ott35042, mrcaott4083ott370807, mrcaott4083ott11712,
## mrcaott4083ott52094, mrcaott4083ott95949, Erythrura ott465905,
## mrcaott24017ott389884, mrcaott24017ott24025, mrcaott24025ott596770,
## mrcaott24025ott453058, mrcaott24025ott141501, mrcaott141501ott966119,
## mrcaott141501ott865472, mrcaott141501ott389883, mrcaott105913ott311555,
## mrcaott311555ott1082386, mrcaott311555ott445491, Taeniopygia
## ott708325, mrcaott4088ott95302, mrcaott4088ott8371, mrcaott4088ott5616,
## mrcaott5616ott28339, mrcaott5616ott6023, mrcaott6023ott243614,
## mrcaott6023ott101225, mrcaott6023ott125079, mrcaott125079ott463026,
## mrcaott125079ott765405, Zonotrichia (genus in domain Eukaryota) ott789032,
## mrcaott125079ott265547, mrcaott125079ott265554, mrcaott5620ott254662,
## mrcaott5620ott29804, mrcaott29804ott449580, mrcaott29804ott449562,
## mrcaott29804ott86894, mrcaott29804ott93045, mrcaott93045ott264496,
## mrcaott264496ott264500, mrcaott264500ott3597163, mrcaott264500ott283668,
## Quiscalus ott743410, mrcaott283673ott673386, mrcaott283673ott741944,
## mrcaott283673ott735243, mrcaott213448ott213452, mrcaott213448ott1009279,
## mrcaott213448ott213451, mrcaott213454ott430627, mrcaott430627ott498751,
## mrcaott430627ott617797, mrcaott430627ott3597159, mrcaott99175ott364331,
## Xanthocephalus ott364336, mrcaott6366ott28332, mrcaott6366ott88283,
## mrcaott6366ott341465, mrcaott6366ott157599, mrcaott6366ott178457,
## mrcaott6366ott28346, mrcaott6366ott238142, mrcaott6366ott6375,
## mrcaott6375ott119724, mrcaott6375ott238137, mrcaott238137ott328909,
## mrcaott238137ott464865, Haemorhous ott3601758, mrcaott9416ott840030,
## mrcaott9416ott96147, mrcaott9416ott407760, Passer ott515158,
## mrcaott9416ott407769, mrcaott9416ott25628, mrcaott9416ott407764,
## mrcaott9416ott68955, mrcaott9416ott73636, mrcaott1488ott72472,
## mrcaott1488ott2375, mrcaott1488ott63797, mrcaott1488ott62251,
## mrcaott1488ott110316, mrcaott1488ott5819, mrcaott1488ott44217,
## mrcaott1488ott16185, Hirundinidae ott897681, mrcaott16185ott67916,
## mrcaott67916ott67921, mrcaott67916ott119537, mrcaott67916ott67920,
## Delichon ott922719, mrcaott107463ott353520, mrcaott107463ott177058,
## mrcaott107463ott107476, mrcaott107476ott326781, mrcaott107476ott337752,
## mrcaott337752ott865537, mrcaott337752ott337762, mrcaott4820ott409116,
## mrcaott4820ott17162, mrcaott4820ott20998, mrcaott4820ott20989,
## mrcaott4820ott58860, mrcaott4820ott23690, mrcaott4820ott20996,
## mrcaott4820ott11462, mrcaott4820ott140440, mrcaott4820ott197505,
## mrcaott4820ott75978, mrcaott4820ott11315, mrcaott4820ott5933,
## mrcaott5933ott60465, mrcaott5933ott25637, mrcaott25637ott199843,
## mrcaott25637ott473431, mrcaott25637ott686166, mrcaott25637ott111993,
## mrcaott25637ott183621, mrcaott183621ott501241, mrcaott501241ott3598010,
## mrcaott501241ott597018, mrcaott686165ott686168, mrcaott60456ott662804,
## mrcaott60456ott75990, Pica ott776480, Falconidae ott212186,
## mrcaott47588ott225286, mrcaott47588ott748842, mrcaott47588ott201377,
## mrcaott201377ott773020, mrcaott179290ott352521, mrcaott179290ott624976,
## mrcaott179290ott624973, mrcaott179290ott624974, mrcaott179290ott432111,
## mrcaott5272ott92263, mrcaott5272ott24121, Scolopacidae ott887699,
## mrcaott24121ott217797, mrcaott24121ott45306, mrcaott24121ott654830,
## mrcaott24121ott214779, mrcaott24121ott651066, mrcaott651066ott1090732,
## mrcaott7639ott306220, mrcaott7639ott57833, mrcaott7639ott383929,
## mrcaott7639ott279504, mrcaott7639ott47401, mrcaott7639ott341047,
## mrcaott7639ott234666, mrcaott7639ott966608, mrcaott7639ott285543,
## Sterna ott16767, mrcaott285543ott341030, mrcaott285543ott738509,
## mrcaott285543ott5560401, mrcaott285543ott966606, mrcaott285543ott966604,
## mrcaott22965ott738512, mrcaott22965ott80679, mrcaott22965ott5859768,
## mrcaott22965ott154126, mrcaott22965ott241571, mrcaott22965ott282132,
## mrcaott22965ott414141, mrcaott22965ott704174, mrcaott22965ott704175,
## mrcaott22965ott324050, mrcaott22965ott353849, mrcaott22965ott75913,
## mrcaott75913ott335736, mrcaott335737ott450107, mrcaott425206ott887691,
## mrcaott425206ott515355, mrcaott119599ott993041, mrcaott119599ott509055,
## mrcaott416082ott4947414, mrcaott147723ott219032, Stercorariidae ott168297,
## Stercorarius ott742632, mrcaott742640ott742641, mrcaott57823ott57827,
## mrcaott57823ott242771, mrcaott57823ott112937, mrcaott112937ott129402,
## mrcaott112937ott341980, mrcaott112937ott112941, mrcaott112941ott112948,
## mrcaott112948ott5341357, mrcaott112948ott242776, mrcaott242776ott313115,
## mrcaott242776ott704464, mrcaott242776ott331994, mrcaott331994ott413796,
## mrcaott9830ott86672, mrcaott9830ott90560, mrcaott9830ott324158,
## mrcaott9830ott55044, mrcaott9830ott285638, mrcaott9830ott117726,
## Sulidae ott452462, mrcaott403772ott429615, Sula ott160486,
## mrcaott429615ott5859716, mrcaott429615ott827441, mrcaott429615ott1030312,
## Procellariiformes ott452461, mrcaott18206ott31011, mrcaott31011ott71459,
## Diomedeidae ott85277, mrcaott71459ott320282, Phoebastria ott941509,
## mrcaott320282ott320284, Sphenisciformes ott494366, mrcaott60413ott60417,
## mrcaott60413ott429618, Eudyptes ott494364, mrcaott82219ott116946,
## mrcaott116946ott219197, mrcaott134466ott494361, Pygoscelis ott494365,
## mrcaott134466ott783352, mrcaott5021ott198671, mrcaott5021ott75792,
## Cuculiformes ott212171, mrcaott75792ott212172, mrcaott75792ott3601282,
## mrcaott75792ott119216, mrcaott119216ott212175, mrcaott119216ott169265,
## mrcaott169265ott550039, mrcaott169265ott462546, mrcaott462546ott626955,
## mrcaott462546ott549514, mrcaott549514ott3596314, mrcaott549514ott3596309,
## mrcaott549514ott550040, mrcaott549514ott3596308, mrcaott549514ott3596307,
## mrcaott549514ott3596306, mrcaott549514ott</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(tr, <span class="dt">show.tip.label=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="meta-analysis_files/figure-html/birds_in_a_tree-1.png" width="480" style="display: block; margin: auto;"></p>
<p>There are a few things to note here. First, the tree has no branch lengths. At present this is true for the whole of the Open Tree synthetic tree. Some comparative methods require either branch lengths or an ultrametric tree. Before you can use one of those methods you will need to get a tree with branch lengths. You could try looking for published trees made available by the Open Tree with <code>studies_find_trees</code>. Alternatively, you could estimate branch lengths from the toplogy of a phylogeny returned by <code>tol_induced_subtree</code>, perhaps by downloading DNA sequences from the NCBI with <code>rentrez</code> or “hanging” the tree on nodes of known-age using penalized likelihood method in <code><a href="https://www.rdocumentation.org/packages/ape/topics/chronos">ape::chronos</a></code>. In this case, we will use only the topology of the tree as input to our comparative analysis, so we can skip these steps.</p>
<p>Second, the tip labels contain OTT IDs, which means they will not perfectly match the species names in our dataset or the taxon map that we created earlier:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">tr<span class="op">$</span>tip.label[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</a></code></pre></div>
<pre><code>## [1] "Pteria_hirundo_ott112607"      "Zaus_caeruleus_ott14254"      
## [3] "Ficedula_albicollis_ott107840" "Luscinia_svecica_ott274225"</code></pre>
<p>Finally, the tree contains node labels for those nodes that match a higher taxonomic group, and empty character vectors (<code>""</code>) for all other nodes. Some comparative methods either do no expect node labels at all, or require all labeled nodes to have a unique name (meaning multiple “empty” labels will cause and error).</p>
<p>We can deal with all these details easily. <code>rotl</code> provides the convenience function <code>strip_ott_ids</code> to remove the extra information from the tip labels. With the IDs removed, we can use our taxon map to replace the tip labels in the tree with the species names from dataset.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">otl_tips &lt;-<span class="st"> </span><span class="kw"><a href="../reference/strip_ott_ids.html">strip_ott_ids</a></span>(tr<span class="op">$</span>tip.label, <span class="dt">remove_underscores=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">tr<span class="op">$</span>tip.label &lt;-<span class="st"> </span>taxon_map[ otl_tips ]</a></code></pre></div>
<p>Finally, we can remove the node labels by setting the <code>node.label</code> attribute of the tree to <code>NULL</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">tr<span class="op">$</span>node.label &lt;-<span class="st"> </span><span class="ot">NULL</span></a></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">egg_data &lt;-<span class="st"> </span>egg_data[egg_data<span class="op">$</span>animal <span class="op">%in%</span><span class="st"> </span>tr<span class="op">$</span>tip.label, ]</a></code></pre></div>
</div>
<div id="perform-the-meta-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#perform-the-meta-analysis" class="anchor"></a>Perform the meta-analysis</h3>
<p>Now we have data and a tree, and we know the names in the tree match the ones in the data. It’s time to do the comparative analysis. Rutkowska <em>et al</em>. used <code>MCMCglmm</code>, a Bayesian MCMC approach to fitting multi-level models,to perform their meta-analysis, and we will do the same. Of course, to properly analyse these data you would take some care in deciding on the appropriate priors to use and inspect the results carefully. In this case, we are really interested in using this as a demonstration, so we will just run a simple model.</p>
<p>Specifically we sill fit a model where the only variable that might explain the values of <code>Zr</code> is the random factor <code>animal</code>, which corresponds to the phylogenetic relationships among species. We also provide <code>Zvr</code> as the measurement error variance, effectively adding extra weight to the results of more powerful studies. Here’s how we specify and fit that model with <code>MCMCglmm</code>:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">123</span>)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="cf">if</span> (<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">require</a></span>(MCMCglmm, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)) {</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"></a>
<a class="sourceLine" id="cb19-4" data-line-number="4">    pr &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">R=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">V=</span><span class="dv">1</span>,<span class="dt">nu=</span><span class="fl">0.002</span>),</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">               <span class="dt">G=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">G1=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">V=</span><span class="dv">1</span>,<span class="dt">nu=</span><span class="fl">0.002</span>))</a>
<a class="sourceLine" id="cb19-6" data-line-number="6">               )</a>
<a class="sourceLine" id="cb19-7" data-line-number="7"></a>
<a class="sourceLine" id="cb19-8" data-line-number="8">    model &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/MCMCglmm/topics/MCMCglmm">MCMCglmm</a></span>(Zr<span class="op">~</span><span class="dv">1</span>,<span class="dt">random=</span><span class="op">~</span>animal,</a>
<a class="sourceLine" id="cb19-9" data-line-number="9">                      <span class="dt">pedigree=</span>tr,</a>
<a class="sourceLine" id="cb19-10" data-line-number="10">                      <span class="dt">mev=</span>egg_data<span class="op">$</span>VZr,</a>
<a class="sourceLine" id="cb19-11" data-line-number="11">                      <span class="dt">prior=</span>pr,</a>
<a class="sourceLine" id="cb19-12" data-line-number="12">                      <span class="dt">data=</span>egg_data,</a>
<a class="sourceLine" id="cb19-13" data-line-number="13">                      <span class="dt">verbose=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb19-14" data-line-number="14">} <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb19-15" data-line-number="15">    model &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/readRDS">readRDS</a></span>(<span class="dt">file =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"mcmcglmm_model.rds"</span>, <span class="dt">package =</span> <span class="st">"rotl"</span>))</a>
<a class="sourceLine" id="cb19-16" data-line-number="16">}</a></code></pre></div>
<pre><code>## Warning in inverseA(pedigree = pedigree, scale = scale, nodes = nodes): no
## branch lengths: compute.brlen from ape has been used</code></pre>
<pre><code>## Warning: 'cBind' is deprecated.
##  Since R version 3.2.0, base's cbind() should work fine with S4 objects</code></pre>
<p>Now that we have a result we can find out how much phylogenetic signal exists for sex-biased differences in egg-size. In a multi-level model we can use variance components to look at this, specifically the proportion of the total variance that can be explained by phylogeny is called the phylogenetic reliability, <em>H</em>. Let’s calculate the <em>H</em> for this model:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">var_comps &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colSums">colMeans</a></span>(model<span class="op">$</span>VCV )</a>
<a class="sourceLine" id="cb22-2" data-line-number="2">var_comps[<span class="st">"animal"</span>] <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(var_comps)</a></code></pre></div>
<pre><code>##      animal 
## 0.002399979</code></pre>
<p>It appears there is almost no phylogenetic signal to the data. The relationships among species explain much less that one percent of the total variance in the data. If you were wondering, Rutkowska <em>et al</em>. report a similar result, even after adding more predictors to their model most of the variance in <code>Zr</code> was left unexplained.</p>
</div>
</div>
<div id="what-other-comparative-methods-can-i-use-in-r" class="section level2">
<h2 class="hasAnchor">
<a href="#what-other-comparative-methods-can-i-use-in-r" class="anchor"></a>What other comparative methods can I use in R?</h2>
<p>Here we have demonstrated just one comparative analysis that you might do in R. There are an ever-growing number of packages that allow an ever-growing number of analysis to performed in R. Some “classics” like ancestral state reconstruction, phylogenetic independent contrasts and lineage through time plots are implemented in <code>ape</code>. Packages like <code>phytools</code>, <code>caper</code> and <code>diversitree</code> provide extensions to these methods. The <a href="https://CRAN.R-project.org/view=Phylogenetics">CRAN Phylogenetics Taskview</a> gives a good idea of the diversity of packages and analyses that can be completed in R.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#phylogenetic-comparative-methods">Phylogenetic Comparative Methods</a></li>
      <li><a href="#a-phylogenetic-meta-analysis">A phylogenetic meta-analysis</a></li>
      <li><a href="#what-other-comparative-methods-can-i-use-in-r">What other comparative methods can I use in R?</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Francois Michonneau, Joseph Brown, David Winter.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
